### PR TITLE
Add config option to use GitHub org namespace with single org

### PR DESCRIPTION
**Pull request by stephenglass**
## Hey, I just made a Pull Request!

If you only list a single org in the GitHub org provider config, the generated group entities will use the `default` namespace, otherwise they will use the org name as the namespace. There is no way to configure this behavior from config.

We have multiple GitHub orgs but for our Proof-of-concept app we are consuming from a single org. All the generated group entities will have `default` namespace. In the future, we will want to expand the orgs we consume from but don't want to be left with orphaned entities from the `default` namespace. We also want the group entities namespace to match the org name for demo purposes, which more easily demonstrates the capability of consuming from multi orgs.

This PR adds a new optional config option `neverUseDefaultNamespace` to GitHub provider. If this flag is set, it will always use the org name as the namespace even if only a single org was provided.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by vinzscam — CHANGES_REQUESTED** [review]
When you switch from a single org to multiple orgs there will be no orphaned users left, as the providers will take care of cleaning up the entities.
Does this solve your problem?

**Comment by stephenglass** [comment]
> When you switch from a single org to multiple orgs there will be no orphaned users left, as the providers will take care of cleaning up the entities.
> 
> Does this solve your problem?
> 
> 
> 
> 
> 
> 

Thanks, it solves the problem of the orphaned entities! But what if I just prefer to use the org as namespace instead of default? I suppose you could override the default behavior in a custom transformer but is that better than a config option?

File: docs/integrations/github/org.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ Directly under the `githubOrg` is a list of configurations, each entry is a stru
 - `githubUrl`: The target that this provider should consume
 - `orgs` (optional): The list of the GitHub orgs to consume. If you only list a single org the generated group entities will use the `default` namespace, otherwise they will use the org name as the namespace. By default the provider will consume all accessible orgs on the given GitHub instance (support for GitHub App integration only).
 - `schedule`: The refresh schedule to use, matches the structure of [`TaskScheduleDefinitionConfig`](https://backstage.io/docs/reference/backend-tasks.taskscheduledefinitionconfig/)
+- `neverUseDefaultNamespace` (optional): Generated group entities will use the org name as the namespace, even if only a single org is provided.
```
Comment (freben) [inline docs/integrations/github/org.md:99]: Could we make this `namespacePerOrg: true` instead? I feel it's clearer, and also it makes better room for a future `namespace: 'default'` to force it the other way (using a single namespace for multiple orgs being pulled in).

**Review by freben — COMMENTED** [review]
Terribly sorry that this review took so long to get to. Looks good, just one suggestion!

File: docs/integrations/github/org.md (hunk 1 above)
Comment (stephenglass) [reply docs/integrations/github/org.md]: Done!

**Comment by vinzscam** [comment]
> > When you switch from a single org to multiple orgs there will be no orphaned users left, as the providers will take care of cleaning up the entities.
> > Does this solve your problem?
> 
> Thanks, it solves the problem of the orphaned entities! But what if I just prefer to use the org as namespace instead of default? I suppose you could override the default behavior in a custom transformer but is that better than a config option?

You are correct in saying that using a config is easier than using a custom transformer. However, we need to be mindful about the features we add, as this will entail extra costs in maintenance and code size. Given that I don't believe many adopters will utilize this feature (you also mentioned that this would be a temporary feature for you until you transition to using multiple organizations), and considering that a custom transformer can help you achieve this, I don't see the need to add this extra feature.

**Comment by stephenglass** [comment]
> > > When you switch from a single org to multiple orgs there will be no orphaned users left, as the providers will take care of cleaning up the entities.
> 
> > > Does this solve your problem?
> 
> > 
> 
> > Thanks, it solves the problem of the orphaned entities! But what if I just prefer to use the org as namespace instead of default? I suppose you could override the default behavior in a custom transformer but is that better than a config option?
> 
> 
> 
> You are correct in saying that using a config is easier than using a custom transformer. However, we need to be mindful about the features we add, as this will entail extra costs in maintenance and code size. Given that I don't believe many adopters will utilize this feature (you also mentioned that this would be a temporary feature for you until you transition to using multiple organizations), and considering that a custom transformer can help you achieve this, I don't see the need to add this extra feature.

Totally understood and I agree. I'll close this PR. Appreciate your reviews as always!

**Pull request closed without merging**